src/emscatter.jl

```diff
@@ -0,0 +1,307 @@
+# 
+# Since PML absorbs all waves in $x(y)$ direction, the associated boundary condition is then usually the zero Dirichlet boundary condition. Here, the boundary conditions are zero Dirichlet boundary on the top and bottom side $\Gamma_D$ but periodic boundary condition on the left ($\Gamma_L$) and right side ($\Gamma_R$). 
+# The reason that we use a periodic boundary condition for the left and right side instead of zero Dirichlet boundary condition is that we want to simulate a plane wave exicitation, which then requires a periodic boundary condition.
+#
+# Consider $\mu(x)=1$ (which is mostly the case in electromagnetic problems) and denote $\Lambda=\frac{1}{1+\mathrm{i}\sigma(u)/\omega}$, we can formulate the problem as 
+#
+# ```math
+# \left\{ \begin{aligned} 
+# \left[-\Lambda\nabla\cdot\frac{1}{\varepsilon(x)}\Lambda\cdot\nabla -k^2\right] H &= f(x) & \text{ in } \Omega,\\
+# H&=0 & \text{ on } \Gamma_D,\\
+# H|_{\Gamma_L}&=H|_{\Gamma_R},&\\
+# \end{aligned}\right.
+# ```
+# 
+# # Numerical scheme
+# 
+# Similar to the previous tutorials, we need to construct the weak form for the above PDEs. After integral by part and removing the zero boundary integral term, we get:
+#
+# ```math
+# a(u,v) = \int_\Omega \left[\nabla(\Lambda v)\cdot\frac{1}{\varepsilon(x)}\Lambda\nabla u-k^2uv\right]\mathrm{d}\Omega
+# ```
+#
+# ```math
+# b(v) = \int_\Omega vf\mathrm{d}\Omega
+# ```
+#
+# # Setup
+# 
+# We import the packages that will be used, define the geometry and physics parameters. 
+# 
+
+using Gridap
+using GridapGmsh
+using Gridap.Fields
+using Gridap.Geometry
+
+# Geometry parameters
+λ = 1.0          # Wavelength (arbitrary unit)
+L = 4.0          # Width of the area
+H = 6.0          # Height of the area
+xc = [0 -1.0]    # Center of the cylinder
+r = 1.0          # Radius of the cylinder
+d_pml = 0.8      # Thickness of the PML   
+# Physical parameters 
+k = 2*π/λ        # Wave number 
+ϵ1 = 3.0         # Relative electric permittivity for cylinder
+
+# # Discrete Model
+# 
+# We import the model from the `geometry.msh` mesh file using the `GmshDiscreteModel` function defined in `GridapGmsh`. The mesh file is created with GMSH in Julia (see the file ../assets/emscatter/MeshGenerator.jl). Note that this mesh file already contains periodic boundary information for the left and right side, and that is enough for gridap to realize a periodic boundary condition should be implemented. 
+# 
+
+model = GmshDiscreteModel("../models/geometry.msh")
+
+# # FE spaces
+# 
+# We use the first-order lagrangian as the finite element function space basis. The dirihlet edges are labeld with `DirichletEdges` in the mesh file. Since our problem involves complex numbers (because of PML), we need to assign the `vector_type` to be `Vector{ComplexF64}`.
+# 
+
+# Test and trial finite element function space
```

Comment:
Use a h3 level header here, namely `# ### Test and trial finite element function space`